civis/ml/_model.py

```diff
@@ -450,3 +462,518 @@ def training_metadata(self):
+        A manifest file is the output of a Civis
+        export job with ``force_multifile=True`` set,
+        e.g. from :func:`civis.io.civis_to_multifile_csv`.
+        Large Civis Tables (provided using ``table_name``)
+        will automatically be exported to manifest files.
+
+        Prediction outputs will always be stored as gzipped
+        CSVs in one or more Civis Files. You can find a list of
+        File ID numbers for output files at the "output_file_ids"
+        key in the metadata returned by the prediction job.
+        Provide an ``output_table`` (and optionally an ``output_db``,
+        if it's different from ``database_name``) to copy these
+        predictions into a Civis Table.
+
+        Parameters
+        ----------
+        X : pd.DataFrame or str, optional
+            Local data, as either a ``DataFrame`` or as the
+            file name of a CSV. Note that if this is a ``DataFrame``,
+            the index will be ignored.
+        table_name : str, optional
+            The qualified name of the table containing your data
+        database_name : str, optional
+            Name of the database holding the
+            data, e.g., 'My Redshift Cluster'.
+        manifest : int, optional
+            ID for a manifest file stored as a Civis file.
+            (Note: if the manifest is not a Civis Platform-specific manifest,
+            like the one returned from `civis.io.civis_to_multfile_csv`,
+            this must be used in conjunction with table_name and database_name
+            due to the need for column discovery via Redshift.)
+        file_id : int, optional
+            If the data are a CSV stored in a Civis file,
+            provide the integer file ID.
+        sql_where : str, optional
+            A SQL WHERE clause used to scope the rows to be predicted
+        sql_limit : int, optional
+            SQL LIMIT clause to restrict the size of the prediction set
+        primary_key : str, optional
+            Primary key of the prediction table. Defaults to
+            the primary key of the training data.
+        output_table: str, optional
+            The table in which to put the predictions.
+        output_db : str, optional
+            Database of the output table. Defaults to the database
+            of the input table.
+        if_exists : {'fail', 'append', 'drop', 'truncate'}
+            Action to take if the prediction table already exists.
+        n_jobs : int, optional
+            Number of concurrent Platform jobs to use
+            for multi-file / large table prediction.
+        polling_interval : float, optional
+            Check for job completion every this number of seconds.
+            Do not set if using the notifications endpoint.
+
+        Returns
+        -------
+        :class:`~ModelFuture`
+        """
+        if self.train_result_.running():
+            raise RuntimeError('Wait for the training to finish.')
```

Comment:
Do you have a use case in mind for raising an error here?  I think it would be okay to simply block until the model finished training.